inflate.c

```diff
@@ -216,17 +216,54 @@ static int32_t updatewindow(PREFIX3(stream) *strm, const uint8_t *end, uint32_t
 
     /* copy state->wsize or less output bytes into the circular window */
     if (copy >= state->wsize) {
-        memcpy(state->window, end - state->wsize, state->wsize);
+        /* Only do this if the caller specifies to checksum bytes AND the platform requires
+         * it (s/390 being the primary exception to this. Also, for now, do the adler checksums
+         * if not a gzip based header. The inline adler checksums will come in the near future,
+         * possibly the next commit */
+        if (INFLATE_NEED_CHECKSUM(strm) && (state->wrap & 4)) {
+#ifdef GUNZIP 
+            /* We have to split the checksum over non-copied and copied bytes */
+            if (copy > state->wsize)
+                functable.crc32_fold(&state->crc_fold, end - copy, copy - state->wsize, 0);
+            functable.crc32_fold_copy(&state->crc_fold, state->window, end - state->wsize, state->wsize);
```

Comment:
There is no adler checksum being done `if (state->flags)`. 
```
if (state->flags) {
   if (copy > state->wsize)
      functable.crc32_fold(&state->crc_fold, end - copy, copy - state->wsize, 0);
   functable.crc32_fold_copy(&state->crc_fold, state->window, end - state->wsize, state->wsize);
} else {
   strm->adler = state->check = adler32(state->check, end - copy, copy);
   memcpy(state->window, end - state->wsize, state->wsize);
}
```

Reply:
https://github.com/madler/zlib/blob/cacf7f1d4e3d44d871b605da3b647f07d718623f/inflate.c#L449 might be helpful to look at..

Reply:
Ahhh, I had missed that subtlety in that macro.  Though, this should basically compile to the same thing, right?

Reply:
What you have right now would not compile to the same thing, because you are missing the adler32 branch when GZUNZIP is defined.

Reply:
Oh I see, I should be able to get to this later tonight, hopefully that's all that's preventing this from working.

Reply:
Hooray, working now.

So is the #ifdef GUNZIP macro there mostly as an optimization for when compiling with minigzip or something?

Reply:
This is what I found in `inflate.h`..
<img width="537" alt="image" src="https://user-images.githubusercontent.com/1364432/158290211-a275ff7a-5974-43a5-bea2-93a0e9b6579f.png">

Reply:
@Dead2 maybe we should get rid of this `GUNZIP` stuff...

Reply:
@nmoinvaz It would potentially hurt applications that compile zlib-ng statically and only ever want to use the deflate wrapper for example. How much? probably not much.
It could be worth doing to simplify maintaining the code though.

Reply:
I've decluttered this a bit so it's not so much of an issue.  We can probably get by without axing it for now.